test/forward_backward_compatibility/check_forward_backward_compatibility.py

```diff
@@ -139,6 +137,45 @@ def allow_listed(schema):
     ("dist_c10d", datetime.date(2099, 9, 17)),
 ]
 
+def has_valid_upgraders(schema, version_map):
+    # we want to parse through the map to find if
+    # the schema has valid upgraders. Since the
+    # version map has entry for each overload
+    # we need to do some ugly parsing.
+
+    # the name of the operator
+    schema_name = schema.name
+
+    if schema_name not in version_map:
+        return False
+
+    entries = version_map[schema_name]
+
+    possible_overloads = []
+    possible_schemas = []
+    for key, upgrader_schema_entries in entries.items():
+        possible_overloads.append(key)
+        possible_schemas.extend(upgrader_schema_entries)
+
+    # let's make sure this existing schema is part of possible
+    # schemas
+    found = False
+    for old_schema in possible_schemas:
+        if old_schema == schema:
+            found = True
+
+    if found:
+        return True
+
+    # this existing schema can be up to date
+    if not found:
+        current_version = torch._C._get_max_operator_version()
+        for overload in possible_overloads:
```

Comment:
Why do we need this for loop actually? If there is no upgrader (`if not found`) for it, then this `_is_op_symbol_current` check will always return `True` right?

Reply:
Yeah true. This is not needed.